src/Types/Grouping.ts

```diff
@@ -0,0 +1,3 @@
+export class Grouping<TKey, TElement> extends Array<TElement> {
+    public key!: TKey;
+}
```

Comment:
Ah yes an object with extra steps

Reply:
I always avoided implementing this because I didn't wanna look up the definition of `IGrouping`. So I finally looked up the definition, what happened next will leave you speechless...

From [the docs](https://docs.microsoft.com/en-us/dotnet/api/system.linq.igrouping-2?view=netcore-3.1#remarks)
> An `IGrouping<TKey,TElement>` is an `IEnumerable<T>` that additionally has a key.

Reply:
💀💀💀

idk why I was surprised but I was